src/Kestrel.Core/Internal/Http2/FlowControl/InputFlowControl.cs

```diff
@@ -90,10 +92,7 @@ public bool TryUpdateWindow(int bytes, out int updateSize)
 
         public void StopWindowUpdates()
         {
-            lock (_flowLock)
-            {
-                _windowUpdatesDisabled = true;
-            }
+            _windowUpdatesDisabled = true;
```

Comment:
Why are we removing the lock here?

Reply:
It didn't do anything

Reply:
To be more specific, this is only used to stop sending window updates to the client after the associated stream or connection is aborted which is a terminal state. If we happen to send an unneeded window update it's not a big deal. This is simply an optimization.

This is only checked in one place, and doesn't need to be synchronized with any other state. _flow is updated whether or not window updates are disabled. I don't like unnecessary code that looks like it might be important but isn't.